src/DataReduction/PCA.jl

```diff
@@ -0,0 +1,41 @@
+# Leverage PCA from MultivariateStats.jl ?
+
+function matricize(VoV::Vector{Vector{FT}}) where {FT}
+    Matrix(reduce(hcat,VoV)')
+end
+
+mutable struct PCA{FT,IT} <: AbstractDRProblem
+    snapshots::Union{Vector{Vector{FT}},Matrix{FT}}
+    nmodes::IT
+    rbasis::Vector{Vector{FT}}
+    energy::FT
+
+    function PCA(snaps::Vector{Vector{FT}},nmodes::IT) where {FT,IT}
+        errorhandle(matricize(snaps),nmodes)
+        new{eltype(snaps[1]),typeof(nmodes)}(snaps,nmodes)
+    end
+
+    function PCA(snaps::Matrix{FT},nmodes::IT) where {FT,IT}
+        errorhandle(snaps,nmodes)
+        new{eltype(snaps),typeof(nmodes)}(snaps,nmodes)
+    end
+end
+
+function reduce!(pca::PCA{FT,IT}) where {FT,IT}
+    op_matrix = pca.snapshots
+    if typeof(pca.snapshots) == Vector{Vector{FT}}
+        op_matrix = matricize(pca.snapshots)
+    end
+    u,s,v = svd(op_matrix)
```

Comment:
In cases where applying POD (model order reduction in general) makes sense, taking the SVD of the whole data matrix is often prohibitively expensive. So I think we should default to using the method of snapshots to compute the dominant singular vectors/values. If the matrix is small it makes hardly a difference. In general it would be good to allow explicit specification of the algorithm to be used. Many options are conceivable: Randomized SVD (see RandomizedLinAlg.jl for example), Lancosz  bidiagonalization (see TSVD.jl).

Reply:
That is an excellent suggestion, I shall implement this!